### ISSUE TITLE
Harvester bug: flags becoming skulls!

### ISSUE THREAD

**Issue by The-Gig** [OpenArena/gamecode#76]
Hi! I just noticed a couple of bugs in Harvester (g_gametype 7).

The first one is so insignificant that maybe it isn't worth a dedicated "issue"... if you start a match and suicide with /kill immediately afterwards, the skull/cube of your death doesn't spawn, then for later suicides it seems to start working correctly. I'm not sure if sometimes it may happen also later or not, however at the start of the match is clearly visible, especially if you are alone in the server.

The second one is a bit more worrisome: for some reason, you can grab red/blue/white flags as if they were skulls/cubes: if you run over one of them, you can see your "owned skull counter" in lower right corner goes up 1 (and in thirdperson you can see a skull following you), so if you then go to the enemy obelisk, you can score even if you are the only player in the game.
Although the white flag is automatically *hidden* in this gametype, it can still be touched and collected as a skull.
It happens only once per match, as the flags don't respawn.

So, for the moment, it's important that map designers use gametype/!gametype keys to prevent red/blue/neutral flag from spawning in harvester mode.
I guess for a server admin/local player point of view, the current workaround would be to **temporarily** 
`set disable_team_CTF_redflag 1`
`set disable_team_CTF_blueflag 1`
`set disable_team_CTF_neutralflag 1`
just before loading a map suffering the problem in harvester, and then `unset` them to do not break the rest of the game.

**Comment by NeonKnightOA** [comment]
Well, flags shouldn't spawn at all in Harvester, so that's indeed a problem.

**Comment by NeonKnightOA** [comment]
I've just submitted a fix. It's #77 .

**Comment by NeonKnightOA** [comment]
While working on #77 I also figured out that other gametypes with their respective key items may be affected as well by this bug, so I did #78 .

**Comment by NeonKnightOA** [comment]
Here's a test pk3: https://cdn.discordapp.com/attachments/405518544382590987/669554355392741389/z_oax_v2.pk3
#78 has also been updated with this pk3's latest code.

**Comment by The-Gig** [comment]
I tried a very quick Harvester test with your oax pk3 above, and it looks like that, even if the red and blue flags are now hidden, they can still be collected as skulls if you go to their position.

**Comment by sago007** [comment]
@The-Gig I have been unable to reproduce it. I do not get a skull when running over there the flag used to be. Which map did you test on?

I can see in the code that it is possible and I am preparing a fix for it.

**Comment by The-Gig** [comment]
I was using the first alpha of Toyotafankk ctf1ish... however I guess it should happen with any map where thenflags do not have gametype/!gametype keys set.
Also, if you have followed Discord in the last few days, some further change by NK did actually fix that bug, but introduced a strange small glitch in ctfE mode (wasn't committed to master yet... tested using a test pk3 by nk), with the enemy flag respawning if you ran on it before the match starts.

**Comment by NeonKnightOA** [comment]
Unfortunately while testing the AI, I've found a series of errors. Namely: the flag bug is still there. It appears in Harvester and Overload.

**Comment by sago007** [comment]
I have created a simpler solution for the problem: https://github.com/OpenArena/gamecode/commit/4a55a1c808e331899be8a5539ba91fd3660fd71a but testing it is a bit difficult.

I do read Discord but forget almost everything from there. I find it an annoying way to keep track of things.

**Comment by The-Gig** [comment]
I've done some tests with it the current nigthly builds version https://files.poulsander.com/~poul19/public_files/oa/dev088/gamecode_nightly/openarena_gamecode_2020-01-25_15a04f8.zip and it looks like the harvester "grabbable flags" bugs is solved there, and it doesn't show the ctfE glitch. Good!

*I have to do some more testing*, but for the moment it looks like oax isn't affected by the small "using /kill right after the match start doesn't spawn a skull" anomaly. Did you make some change which might have fixed it, too? Maybe I should test it in some previous oax version, maybe it has been fixed with some previous change?

**Comment by NeonKnightOA** [comment]
So, in that case, I have to close and discard my solution.

**Comment by The-Gig** [comment]
Sorry for making you wait, but now I've finally done my additional test. It looks like everything is fixed, even the "skull doesn't spawn if you suicide right after match start" thing (I don't know when it has been fixed exactly... I tried an OAX dated december 2018 and it was still there. Well, that's not important.).
I tried with this version https://files.poulsander.com/~poul19/public_files/oa/dev088/gamecode_nightly/openarena_gamecode_2020-02-04_2fef72b.zip and it looks okay to me.

So, closing this issue and *thank you guys!*
